File: Core/Entities/Item.cs
using Newtonsoft.Json;

namespace Core.Entities
{
    public class Item : BaseEntity
    {
        public required string Name { get; set; }
        public required string Description { get; set; }
        public required string Category { get; set; }
        public List<Photo>? Photos { get; set; } = [];
        public List<Pdf>? Pdfs { get; set; } = [];
        public List<Zip>? Zips { get; set; } = [];
        public Editor? Editor { get; set; }

        public DateTime DateCreated { get; set; } = DateTime.UtcNow;
        public int DownloadCount { get; set; } = 0;   
        

        public int FolderId { get; set; }
        public Folder? Folder { get; set; }
    }
}

File: frontend/src/app/shared/models/item.ts
import {Pdf} from './pdf';
import {Zip} from './zip';
import {Editor} from './editor';

export interface Item {
  name: string
  description: string
  category: string
  photos?: Photo[]
  pdfs?: Pdf[]
  zips?: Zip[]
  editor?: Editor
  dateCreated: Date
  downloadCount: number
  id: number
  folderId: number
  isPublic: boolean
  appUserId: string;
}

export interface Photo {
  url: string
  publicId?: string
  itemId: number
  id: number;
}
